### PR TITLE
Update jobs based on s2i to use clonerefs and initupload

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @bparees @gabemontero

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
A job only calls out the repos it needs to sync if they are not code under test already.

File: sjb/config/common/test_cases/s2i.yml [hunk 1]
```diff
@@ -14,3 +14,38 @@ actions:
...
+      trap 'exit 0' EXIT
+      if [[ -n "${JOB_SPEC:-}" ]]; then
+        if [[ "$( jq --compact-output ".buildid" <<<"${JOB_SPEC}" )" =~ ^\"[0-9]+\"$ ]]; then
+          echo "Keeping BUILD_ID"
+        else
+          echo "Using BUILD_NUMBER"
+          JOB_SPEC="$( jq --compact-output ".buildid |= \"${BUILD_NUMBER}\"" <<<"${JOB_SPEC}" )"
+        fi
+        docker run -e JOB_SPEC="${JOB_SPEC}" -v "/data:/data:z" registry.svc.ci.openshift.org/ci/gcsupload:latest --dry-run=false --gcs-bucket=origin-ci-test --gcs-credentials-file=/data/credentials.json --path-strategy=single --default-org=openshift --default-repo=origin /data/gcs/*
+      fi
```
Comment (bparees) [inline sjb/config/common/test_cases/s2i.yml]: it's so much simpler now........

Comment (stevekuznetsov) [reply sjb/config/common/test_cases/s2i.yml]: you weren't pushing any post artifacts in the past, so ...

Comment (stevekuznetsov) [reply sjb/config/common/test_cases/s2i.yml]: we could have not done this but i guess it's nice to have a record of the job actually existing somewhere?

File: sjb/config/test_cases/test_branch_jenkins_images.yml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 ---
 parent: 'common/test_cases/rhel_base_images.yml'
 extensions:
-  sync_repos:
-    - name: "jenkins"
```
Comment (bparees) [inline sjb/config/test_cases/test_branch_jenkins_images.yml]: how does the jenkins repo get synced now?

File: sjb/config/common/test_cases/s2i.yml [hunk 3]
```diff
@@ -4,8 +4,8 @@ provision:
   os: "rhel"
   stage: "base"
   provider: "aws"
-sync_repos:
-    - name: "source-to-image"
+sync:
+  - "openshift,source-to-image=master"
```
Comment (bparees) [inline sjb/config/common/test_cases/s2i.yml]: why do we need to sync openshift here?

**Pull request merged**